db/db_impl/db_impl_open.cc

```diff
@@ -1314,6 +1327,51 @@ Status DBImpl::RecoverLogFiles(const std::vector<uint64_t>& wal_numbers,
   return status;
 }
 
+void DBImpl::MoveCorruptedWalFiles(std::vector<uint64_t>& wal_numbers,
+                                   uint64_t corrupted_wal_number) {
+  size_t corrupt_index_start = wal_numbers.size() - 1;
+  size_t i = 0;
+
+  // Find the index of first corrupted wal.
+  while (i <= wal_numbers.size() - 1) {
+    if (wal_numbers[i] == corrupted_wal_number) {
+      corrupt_index_start = i;
+      break;
+    }
+    i++;
+  }
+
+  std::string archivalPath =
+      ArchivalDirectory(immutable_db_options_.GetWalDir());
+  Status create_status = env_->CreateDirIfMissing(archivalPath);
+
+  // Move all the WAL files from corrupted_wal_number to last WAL
+  // (max_wal_number) to avoid column family inconsistency error to archival
+  // directory. If its unable to create archive dir, it will delete the
+  // corrupted WAL files.
+  while (i <= wal_numbers.size() - 1) {
```

Comment:
I tried to keep the first corrupted file  but I still get the inconsistency error. So currently I moved all corrupted files to an archive folder.

Reply:
Weird. `kPointInTimeRecovery` should be able to proceed to the most-recently-created-and-synced WAL.
What is the cause for the failure?

Reply:
Any findings?